### ISSUE TITLE
MD044 flags words in markdown links

### ISSUE THREAD

**Issue by ithier** [DavidAnson/markdownlint#443]
The new implementation of MD044 is also flagging proper names in links. e.g.

One might want the following capitalization for GPS:

```
MD044:
  names: [
    'GPS',
    ]
```

But then, the following would be flagged as an error:

```
[Link](configuring-the-gps.md)
```

It looks like the previous version excluded links, which didn't cause this problem: https://github.com/DavidAnson/markdownlint/compare/v0.23.1...v0.24.0#diff-504a7cb2d3155e72bce3746e239ad5d6e4c9836c8e87cf365d0a7a0cb93a6b97L21

**Comment by DavidAnson** [comment]
Is this a widespread problem, or is it just this one case? I agree that capitalization of proper names in links can be different, but it doesn't have to be. That link could use capital letters or you could add that link text as a proper name (ex: "configuring-the-gps.md"). Some scenarios may want the new behavior, so I'd have to add a parameter to disable it. But if this is very rare, I'd be curious what you think about the two options above.

**Comment by ithier** [comment]
I've encountered this problem in a MkDocs project which has about 15-20 instances of this problem.

Since the file names of the articles are used in URLs when the documents are hosted, it's not so ideal to have proper name capitalization. Hyphenated lowercase is more or less the convention for URLs.

Adding the link text as a proper name is a manageable work-around, but leads to a tedious workflow and "feels" hacky IMO. Some kind of parameter (like the existing `code_blocks` one) would be nicer and give a better user experience.

**Comment by okalachev** [comment]
Also #435.

MD044 should not check in links and in HTML, I believe.

**Comment by DavidAnson** [comment]
This should have been fixed by https://github.com/DavidAnson/markdownlint/commit/72543a82e76b46c54c3ac00f6e0a0fa447832ba6 for version `0.24.0`. For example, [the example above does not report a violation](https://dlaa.me/markdownlint/#%25m%23%20Issue%20443%0A%0A%5BLink%5D(configuring-the-gps.md)%0A%0A%3C!--%20markdownlint-configure-file%20%7B%0A%20%20%22proper-names%22%3A%20%7B%0A%20%20%20%20%22names%22%3A%20%5B%0A%20%20%20%20%20%20%22GPS%22%0A%20%20%20%20%5D%0A%20%20%7D%0A%7D%20--%3E%0A).
